pkg/controller/container-runtime-config/helpers.go

```diff
@@ -313,9 +317,9 @@ func createCRIODropinFiles(cfg *mcfgv1.ContainerRuntimeConfig) []generatedConfig
 			glog.V(2).Infoln(cfg, err, "error updating user changes for log-level to crio.conf.d: %v", err)
 		}
 	}
-	if ctrcfg.PidsLimit != 0 {
+	if ctrcfg.PidsLimit != nil && *ctrcfg.PidsLimit >= 0 {
```

Comment:
Why are negative values silently ignored here? (Or should this rely on `validateUserContainerRuntimeConfig` and not do redundant checks? The previous version didn’t check for `minPidsLimit`.)

Reply:
Yup, this relies on `validateUserContainerRuntimeConfig`. It won't get to this point if the validation fails.

Reply:
@QiWang19 actually this can just check for `nil` as we are using a pointer now. It was checking for `0` initially as that is the unset value of an int. Leave the validation check to `validateUserContainerRuntimeConfig`.

Reply:
already fixed